chain/consensus/actors/atomic-replace/replace.go

```diff
@@ -0,0 +1,264 @@
+package replace
+
+// Sample actor that replaces the cid from one address to the other.
+// This actor is used as an example of how to use the actor execution
+// protocol.
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/cbor"
+	"github.com/filecoin-project/go-state-types/exitcode"
+	actor "github.com/filecoin-project/lotus/chain/consensus/actors"
+	"github.com/filecoin-project/lotus/chain/consensus/hierarchical/atomic"
+	"github.com/filecoin-project/specs-actors/v7/actors/builtin"
+	"github.com/filecoin-project/specs-actors/v7/actors/runtime"
+	cid "github.com/ipfs/go-cid"
+	xerrors "golang.org/x/xerrors"
+)
+
+//go:generate go run ./gen/gen.go
+
+// example "Replace" actor that atomically replaces the cid from one owner
+// to the other.
+
+var _ runtime.VMActor = ReplaceActor{}
+var _ atomic.LockableActor = ReplaceActor{}
+
+// ReplaceState determines the actor state.
+// FIXME: We are using a non-efficient locking strategy for now
+// where the whole map is locked for an atomic execution.
+// We could use a more fine-grained approach, but this actor
+// has illustrative purposes for now. Consider improving it in
+// future iterations.
+type ReplaceState struct {
+	Owners *atomic.LockedState
+}
+
+type Owners struct {
+	M map[string]cid.Cid
+}
+
+// Merge implement the merge strategy to follow for our lockable state
+// in the actor.
+func (o *Owners) Merge(other atomic.LockableState) error {
+	tt, ok := other.(*Owners)
+	if !ok {
+		return xerrors.Errorf("type of LockableState not Owners")
+	}
+
+	for k, v := range tt.M {
+		_, ok := o.M[k]
+		if ok {
+			return xerrors.Errorf("merge conflict. key for owner already set")
+		}
+		o.M[k] = v
+	}
+	return nil
+
+}
+
+func ConstructState() (*ReplaceState, error) {
```

Comment:
It is not so obvious from a naming perspective why Owner's `ConstructState` returns `*ReplaceState`

Reply:
I guess `ConstructState` is a "de-facto standard" at this point. Is the main function that constructs the state of the actor in construction. In this case the state just happens to have a single data structure for owners.

Reply:
ok. I agree